### ISSUE TITLE
[rio] k3s unable to pull images from localhost

### ISSUE THREAD

**Issue by daxmc99** [rancher/k3s#980]
**Version:**
Provide the output from `k3s -v` and provide the flags used to install or run k3s server.
`0.10.0` inside of k3d `1.3.4`

**Describe the bug**
Possible network change between 0.9 and 0.10

**To Reproduce**
Run k3d with 
```
k3d create  -n rio --image rancher/k3s:v0.10.0 --publish 80:80 --publish 443:443  --publish 9443:9443 --publish 9080:9080
rio install --ip-address=127.0.0.1
# may need to also stop traefik
kubectl delete svc -n kube-system traefik
rio run https://github.com/rancher/rio-demo
```

image fails to pull from localhost

**Expected behavior**
This works with k3s v0.9.1

See https://github.com/rancher/rio/issues/602 for more context

**Comment by davidnuzik** [comment]
Fixed in v0.10.2? Or not yet? https://github.com/rancher/k3s/releases/tag/v0.10.2
@erikwilson 
cc: @rancher-max

**Comment by erikwilson** [comment]
Yes, fixed in v0.10.2.

**Comment by rancher-max** [comment]
Looks good on the rio side! Steps I took:
1. Created new droplet in DigitalOcean.
2. ssh'ed into it and Installed k3s v0.10.2: `curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v0.10.2 sh -`
3. Using rio v0.6.0-alpha.2, ran `rio install` into the cluster
4. Ran the demo: `rio run -n gh https://github.com/rancher/rio-demo`.
It built successfully and the endpoint was available and returned the expected result.

**Comment by izaac** [comment]
@rancher-max this works fine in a k3s cluster with external IP and locally with k3d I see now the images being pulled and the build finishes and the pod gets initialized.

I get connection refused from the endpoints locally (installed rio with `rio install --ip-address=127.0.0.1`).
This is the same behavior with k3s v0.9.1.

We just need to confirm this is due to a cluster config limitation when running k3s locally.

**Comment by ShylajaDevadiga** [comment]
```root@ip-172-31-8-229:~# rio install --ip-address=127.0.0.1
Deploying Rio control plane....
...
Welcome to Rio!
Run `rio run -p 80:8080 https://github.com/rancher/rio-demo` as an example
root@ip-172-31-8-229:~# rio run -p 80:8080 https://github.com/rancher/rio-demo
default:zealous-solomon
root@ip-172-31-8-229:~# curl  http://zealous-solomon-v0-default.n90no4.on-rio.io:31987
Hi there, I'm running in Rio
```
Resolved as of k3s v0.11.0-alpha4. Closing issue.
